linux_os/guide/system/software/integrity/fips/enable_fips_mode/oval/shared.xml

```diff
@@ -7,19 +7,62 @@
       <extend_definition comment="Dracut FIPS module is enabled" definition_ref="enable_dracut_fips_module" />
       <extend_definition comment="system cryptography policy is configured" definition_ref="configure_crypto_policy" />
       <criterion comment="check if system crypto policy selection in var_system_crypto_policy in the profile is set to FIPS" test_ref="test_system_crypto_policy_value" />
-      {{% if product in ["ol8"] %}}
-      <criterion comment="check if the kernel boot parameter is configured for FIPS mode"
-      test_ref="test_grubenv_fips_mode" />
-      {{% elif product in ["rhel8"] %}}
+      <criterion comment="Check if argument fips=1 for Linux kernel is present in /etc/kernel/cmdline" test_ref="test_fips_1_argument_in_etc_kernel_cmdline" />
+      {{% if "ol" in product or "rhel" in product %}}
       <criteria operator="OR">
-        <extend_definition comment="Generic test for s390x architecture"
-        definition_ref="system_info_architecture_s390_64" />
-        <criterion comment="check if the kernel boot parameter is configured for FIPS mode"
-        test_ref="test_grubenv_fips_mode" />
+        <criteria operator="AND">
+          <extend_definition comment="Generic test for s390x architecture"
+          definition_ref="system_info_architecture_s390_64" />
+          <criterion comment="Check if argument fips=1 for Linux kernel is present in /boot/loader/entries/.*.conf"
+          test_ref="test_fips_1_argument_in_boot_loader_entries_conf" />
+        </criteria>
+        <criteria operator="AND">
+          <criteria negate="true">
+            <extend_definition comment="Generic test for NOT s390x architecture"
+            definition_ref="system_info_architecture_s390_64" />
+          </criteria>
+          {{% if product in ["ol8", "rhel8"] %}}
+          <criterion comment="check if the kernel boot parameter is configured for FIPS mode"
+          test_ref="test_grubenv_fips_mode" />
+          {{% else %}}
+          <criterion comment="Check if argument fips=1 for Linux kernel is present in /boot/loader/entries/.*.conf"
+          test_ref="test_fips_1_argument_in_boot_loader_entries_conf" />
+          {{% endif %}}
+        </criteria>
       </criteria>
       {{% endif %}}
     </criteria>
   </definition>
+
+  <ind:textfilecontent54_test id="test_fips_1_argument_in_boot_loader_entries_conf"
+  comment="Check if argument fips=1 is present in the line starting with 'options ' in /boot/loader/entries/.*.conf"
+  check="all" check_existence="all_exist" version="1">
+    <ind:object object_ref="object_fips_1_argument_in_boot_loader_entries_conf" />
+    <ind:state state_ref="state_fips_1_argument_in_boot_loader_entries_conf" />
```

Comment:
`state_fips_1_argument_in_boot_loader_entries_conf` and `state_fips_1_argument_in_etc_kernel_cmdline` have the same definition. Could you use the same state for both tests and reduce duplication?

Reply:
Good catch, fixed.